src/prefect/core/task.py

```diff
@@ -47,6 +74,9 @@ class SignatureValidator(type):
     def __new__(cls, name: str, parents: tuple, methods: dict) -> type:
         run = methods.get("run", lambda: None)
         _validate_run_signature(run)
+        if name not in ["Task", "Parameter"]:
```

Comment:
I'm know I'm deep in metaprogramming here, but this will skip any task even named "Task" rather than base tasks. Is there any way to check the actual type? maybe `cls`?

Reply:
Unfortunately not again; this method is the one that creates the type, so name is all that exists at this stage. Is class Task(Task): even valid python?

Reply:
No but:

```python
from prefect import Task as BaseTask:

class Task(BaseTask):
   👿 
```

Reply:
Actually yes it is lol.

But that's fine, I know I'm digging here.

Reply:
Haha yea, and ultimately this is for the user’s benefit more than system functionality.